cookbooks/aws-parallelcluster-environment/recipes/init/backup_internal_use_shared_data.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+#
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+return if on_docker?
+
+case node['cluster']['node_type']
+when 'HeadNode'
+  Chef::Log.info("Internal Shared Storage #{node['cluster']['internal_shared_dirs']}")
+  # For each, backup the data to a temp location
+  node['cluster']['internal_shared_dirs'].each do |dir|
+    bash "Backup #{dir}" do
+      user 'root'
+      group 'root'
+      code <<-EOH
+        mkdir -p /tmp#{dir}
+        rsync -a #{dir}/ /tmp#{dir}
```

Comment:
I suggest as a best practice to preserve whatever attribute of the source files, not only regular (-a option) but also extended ones (-X option).

Reply:
Thanks!

Reply:
I ran into issues with copying when using the -X option, so retaining just -a for now